nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,337 @@
+"""
+Better brain parcellations for Region of Interest analysis
+"""
+import warnings
+import numbers
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Memory
+
+from .. import masking
+from ..input_data import NiftiMapsMasker
+from .._utils import check_niimg, check_niimg_4d
+from ..image import new_img_like, resample_img
+from ..image.image import _smooth_array, threshold_img
+from .._utils.niimg_conversions import concat_niimgs, _check_same_fov
+from .._utils.compat import _basestring
+from .._utils.ndimage import _peak_local_max
+from .._utils.segmentation import _random_walker
+
+
+def _threshold_maps_ratio(maps_img, threshold):
+    """ Automatic thresholding of atlas maps image.
+
+    Considers the given threshold as a ratio to the total number of voxels
+    in the brain volume. This gives a certain number within the data
+    voxel size which means that nonzero voxels which fall above than this
+    size will be kept across all the maps.
+
+    Parameters
+    ----------
+    maps_img: Niimg-like object
+        an image of brain atlas maps.
+    threshold: float
+        If float, value is used as a ratio to n_voxels to get a certain threshold
+        size in number to threshold the image. The value should be positive and
+        within the range of number of maps (i.e. n_maps in 4th dimension).
+
+    Returns
+    -------
+    threshold_maps_img: Nifti1Image object
+        gives us thresholded image.
+    """
+    maps = check_niimg(maps_img)
+    n_maps = maps.shape[-1]
+    if not isinstance(threshold, float) or threshold <= 0 or threshold > n_maps:
```

Comment:
Maybe you want to use numbers.Real rather than float since threshold can be an int.